### PR TITLE
SAK-43698 - Initial MVP of tool keyset URL

### PR DESCRIPTION
**Pull request by csev**
This is the first of several commits to the branch.   This is MVP of the feature. The next branches will pretty up the UI bits.

### CODE REVIEW

File: basiclti/basiclti-common/src/java/org/sakaiproject/basiclti/util/SakaiBLTIUtil.java [hunk 1]
```diff
@@ -1092,16 +1133,20 @@ public static DeepLinkResponse getDeepLinkFromToken(Map<String, Object> tool, St
 			if (lti_errorlog != null) {
 				log.warn(lti_errorlog);
 			}
-
+/*
+			// TODO: Remove this
```
Comment (ottenhoff) [inline basiclti/basiclti-common/src/java/org/sakaiproject/basiclti/util/SakaiBLTIUtil.java:1137]: just delete this code?

File: basiclti/basiclti-common/src/java/org/sakaiproject/basiclti/util/SakaiBLTIUtil.java [hunk 2]
```diff
@@ -1092,16 +1133,20 @@ public static DeepLinkResponse getDeepLinkFromToken(Map<String, Object> tool, St
...
 			}
 
-			Key publicKey = LTI13Util.string2PublicKey(publicKeyStr);
+			Key publicKey = string2PublicKey(publicKeyStr);
 			if (publicKey == null) {
 				throw new RuntimeException("Could not deserialize tool public key");
 			}
+*/
+			// May throw a RunTimeException on our behalf :)
+		    Key publicKey = SakaiBLTIUtil.getPublicKey(tool, id_token);
```
Comment (ottenhoff) [inline basiclti/basiclti-common/src/java/org/sakaiproject/basiclti/util/SakaiBLTIUtil.java:1149]: indentation looks odd here

File: basiclti/tsugi-util/src/java/org/tsugi/lti13/LTI13KeySetUtil.java [hunk 3]
```diff
@@ -75,4 +77,29 @@ public static String getKeySetJSON(RSAPublicKey key)
...
+		throws java.text.ParseException, com.nimbusds.jose.JOSEException
+	{
+		com.nimbusds.jose.jwk.JWKSet localKeys = com.nimbusds.jose.jwk.JWKSet.parse(json);
+
+        com.nimbusds.jose.jwk.RSAKey nimbusPublic = (com.nimbusds.jose.jwk.RSAKey) localKeys.getKeyByKeyId(kid);
+
+        RSAPublicKey publicKey = nimbusPublic.toRSAPublicKey();
+        java.security.interfaces.RSAKey rsaPublicKey = (java.security.interfaces.RSAKey) publicKey;
+
+		return publicKey;
```
Comment (ottenhoff) [inline basiclti/tsugi-util/src/java/org/tsugi/lti13/LTI13KeySetUtil.java:103]: tabs/spaces

File: basiclti/basiclti-common/src/java/org/sakaiproject/basiclti/util/SakaiBLTIUtil.java (hunk 1 above)
Comment (csev) [reply basiclti/basiclti-common/src/java/org/sakaiproject/basiclti/util/SakaiBLTIUtil.java:1137]: Yup - It was my intent to delete

File: basiclti/basiclti-common/src/java/org/sakaiproject/basiclti/util/SakaiBLTIUtil.java (hunk 2 above)
Comment (csev) [reply basiclti/basiclti-common/src/java/org/sakaiproject/basiclti/util/SakaiBLTIUtil.java:1149]: Yup

**Comment by csev** [comment]
@ottenhoff I am going through a bunch of tests and updating test plans so I won't merge this until I get through that.  I might also find a bug..

**Comment by ottenhoff** [comment]
> @ottenhoff I am going through a bunch of tests and updating test plans so I won't merge this until I get through that. I might also find a bug..

Okay I recommend adding label "Work in Progress" so no one else merges for you

**Pull request merged**